crates/oxc_linter/src/rules/unicorn/prefer_default_parameters.rs

```diff
@@ -0,0 +1,824 @@
+        Expression::AssignmentExpression(assign) => {
+            let target_ok = match &assign.left {
+                AssignmentTarget::AssignmentTargetIdentifier(ident) => {
+                    ident.name.as_str() != param_name
+                }
+                _ => false,
+            };
+            target_ok && is_side_effect_free_expression(&assign.right, param_name)
+        }
+        Expression::BinaryExpression(bin) => {
+            is_side_effect_free_expression(&bin.left, param_name)
+                && is_side_effect_free_expression(&bin.right, param_name)
+        }
+        Expression::UnaryExpression(unary) => {
+            !matches!(unary.operator, oxc_ast::ast::UnaryOperator::Delete)
+                && is_side_effect_free_expression(&unary.argument, param_name)
+        }
+        _ => false,
+    }
+}
+
+fn check_no_extra_references<'a>(
+    ctx: &LintContext<'a>,
+    param_ident_span: Span,
+    param: &FormalParameter<'a>,
+) -> bool {
+    let BindingPatternKind::BindingIdentifier(binding_ident) = &param.pattern.kind else {
+        return false;
+    };
+
+    let Some(symbol_id) = binding_ident.symbol_id.get() else {
+        return false;
+    };
+
+    let references: Vec<_> = ctx.scoping().get_resolved_references(symbol_id).collect();
+
+    if references.len() != 1 {
+        return false;
+    }
+
+    let reference = &references[0];
+    ctx.semantic().reference_span(reference) == param_ident_span
+}
+
+fn check_no_extra_references_assignment<'a>(
+    ctx: &LintContext<'a>,
+    param_ident_span: Span,
+    param: &FormalParameter<'a>,
+) -> bool {
+    let BindingPatternKind::BindingIdentifier(binding_ident) = &param.pattern.kind else {
+        return false;
+    };
+
+    let Some(symbol_id) = binding_ident.symbol_id.get() else {
+        return false;
+    };
+
+    let (has_matching_read, writes) = ctx.scoping().get_resolved_references(symbol_id).fold(
+        (false, 0usize),
+        |(has_matching_read, writes), r| {
+            if r.is_write() {
+                (has_matching_read, writes + 1)
+            } else {
+                let is_matching = ctx.semantic().reference_span(r) == param_ident_span;
+                (has_matching_read || is_matching, writes)
+            }
+        },
+    );
+
+    writes == 1 && has_matching_read
+}
+
+fn check_expression<'a>(
+    ctx: &LintContext<'a>,
+    node: &AstNode<'a>,
+    left_name: &str,
+    right: &Expression<'a>,
+    is_assignment: bool,
+    stmt_span: Span,
+) {
+    let Expression::LogicalExpression(logical_expr) = right.without_parentheses() else {
+        return;
+    };
+
+    if !matches!(logical_expr.operator, LogicalOperator::Or | LogicalOperator::Coalesce) {
+        return;
+    }
+
+    let Expression::Identifier(param_ident) = logical_expr.left.without_parentheses() else {
+        return;
+    };
+
+    let param_name = param_ident.name.as_str();
+
+    if !&logical_expr.right.get_inner_expression().is_literal() {
```

Comment:
```suggestion
    if !logical_expr.right.get_inner_expression().is_literal() {
```

Creating a reference here is unnecessary